src/App.tsx

```diff
@@ -1,223 +1,212 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
 
+import {
+  getTodos,
+  createTodo,
+  deleteTodos,
+  updateTodo,
+} from './api/todos';
+
+import { SortType } from './types/SortType';
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoForm } from './components/TodoForm';
+
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.ALL);
+  const [error, setError] = useState('');
+  const [title, setTitle] = useState('');
+  const [selectedId, setSelectedId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+
+  const filteredTodo = useMemo(() => todos.filter(todo => {
+    switch (sortType) {
+      case SortType.ALL:
+        return todo;
+      case SortType.ACTIVE:
+        return !todo.completed;
+      case SortType.COMPLETED:
+        return todo.completed;
+      default:
+        return null;
+    }
+  }), [todos, sortType]);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
```

Comment:
let's combine these useEffects, both of them works as componentDidMount method
why did you decide to create them in separate way?